bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,338 @@
+The bootable JAR is composed of:
+
+* Zipped server provisioned thanks to Galleon (slim or fat server).
+* User application (WAR, JAR, EAR) unless a hollow bootable JAR is created.
+* Runtime to launch server.
+
+The plugin outputs a JAR named <project>-wildfly.jar
+
+===== Bootable JAR runtime resolution
+
+The set of classes that implement the boot logic are retrieved at packaging time. 
+The new “org.wildfly.core:wildfly-jar-boot” JAR artifact (implementation located in wildfly-core repository) 
+is resolved thanks to the wildfly-core version of the provisioned server and shaded into the bootable JAR. 
+If the artifact doesn’t exist, the packaging will abort, the server being packaged not supporting bootable 
+JAR packaging. (e.g.: old release not containing new artifacts).
+
+This artifact is in charge to unzip the server and setup a JBOSS_HOME. Once this is done, 
+it setups JBoss module classloader and call into a new JBoss module “org.wildfly.bootable-jar” 
+that is in charge of the actual server run.  This new JBoss module implementation located in wildfly-core, 
+(“org.wildfly.core:wildfly-jar-runtime” artifact), is expected to be automatically provisioned by Galleon in all cases. 
+If the module is not part of the provisioned server (for some un-expected reasons), the server execution aborts.
+
+===== Galleon provisioning
+
+Galleon provisions a server thanks to the WildFly Galleon feature-pack.
+Galleon provisioning is operated inside the Maven plugin. There are 2 configuration ways:
+
+* Galleon configuration as Maven plugin configuration items:
+** A Galleon feature-pack location that identifies the server and version. This is required if no provisioning.xml file is provided nor feature-packs list.
+** A list of feature-packs (if no feature-pack location has been set).
+** Optionally, a list of layers to include.
+** Optionally, a list of layers to exclude.
+* Galleon provisioning.xml file located in <app src>/galleon/provisioning.xml. NB: Maven plugin configuration items (if set) override provisioning.xml file. 
+Using provisioning.xml file, custom feature-packs can be combined during the provisioning phase to provision modules, features, content (e.g.: DB drivers).
+
+If no Galleon layers are specified, then a configuration identical to the default _standalone-microprofile.xml_ is provisioned.
+
+===== CLI script execution
+
+In order to have management model changes persisted in the server configuration, the Maven plugin offers a support for CLI script(s) execution. 
+The changes made to the management model during build are persisted in the bootable JAR 
+(as opposed to changes done to the management model at runtime that are lost after a process restart).
+ 
+When building the bootable JAR, a path to a set of CLI scripts can be provided to update the server configuration packaged in the bootable JAR. The CLI scripts are concatenated and executed once the server is provisioned and deployment copied into it (if any).
```

Comment:
Is there some supported way how to write CLI scripts? This seems to be main mean of configuration so it would be a good thing to have a comfortable way how to do it. Although this seems out of scope of this proposal I think it should be evaluated.

Reply:
Using CLI and DMR management operations is very common in WildFly/EAP context. The various admin guides cover intensively DMR operations.

Reply:
They are mentioned everywhere across the documentation but almost always in context of WildFly CLI tool. Here, the user is expected to provide a text files containing sequence of CLI commands. IMHO the usability heavily depends on the way how user can construct such file.

Reply:
Yes, you are right, a chapter to provide details on how the CLI scripts are written and run would help. Adding that.

Reply:
I have updated the CLI part with more details on how to define CLI scripts. The master branch has evolved in order to add more flexibility in the way CLI scripts are configured (I ran into a limitation when adding the web-clustering example, see https://github.com/wildfly-extras/wildfly-jar-maven-plugin/issues/7). The community doc of Alpha4 is no more in sync with respect to the CLI execution configuration, so I added the community doc content in the analysis document.